client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx

```diff
@@ -29,10 +28,14 @@ const CustomDomainPurchaseDetail = ( { selectedSite, hasDomainCredit, translate
 				buttonText={ translate( 'Claim your free domain' ) }
 				href={ `/domains/add/${ selectedSite.slug }` }
 			/>
-			: null;
-
-	const renderHasCustomDomain = () =>
-		<PurchaseDetail
+		);
+	} else if ( ! hasDomainCredit && hasCustomDomain( selectedSite ) ) {
+		const actionButton = {};
```

Comment:
The name here makes it sound like this is a component instead of a grouping of props. Perhaps we should refactor PurchaseDetail a bit.